examples/I/hmat.c

```diff
@@ -0,0 +1,72 @@
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <memkind.h>
+
+#include "hashmap.h"
+
+struct db_data {
+    char* name;
+    void* data;
+};
+
+void* lowest_latency_malloc(size_t size)
+{
+    return memkind_malloc(MEMKIND_LOWEST_LATENCY_LOCAL, size);
+}
+
+void lowest_latency_free(void* ptr)
+{
+    memkind_free(MEMKIND_LOWEST_LATENCY_LOCAL, ptr);
+}
+
+void* highest_capacity_malloc(size_t size)
+{
+    return memkind_malloc(MEMKIND_HIGHEST_CAPACITY, size);
+}
+
+void highest_capacity_free(void* ptr)
+{
+    memkind_free(MEMKIND_HIGHEST_CAPACITY, ptr);
+}
+
+int main(void)
+{
+    const int DB_SIZE = 100;
+    const int NAME_SIZE = 16;
+    const int DATA_SIZE = 10 * 1024;
+    const int BUCKETS_NUM = 100;
+    int i;
+
+    struct alloc_t lowest_latency_allocator = {lowest_latency_malloc, 
+        lowest_latency_free};
+
+    struct hashmap_t* hashmap = hashmap_create(BUCKETS_NUM, 
+        lowest_latency_allocator);
+    
+    struct db_data* db = highest_capacity_malloc(sizeof(struct db_data) * 
+        DB_SIZE);
+
+    for (i = 0; i < DB_SIZE; i++) {
+        db[i].data = highest_capacity_malloc(DATA_SIZE);
+        db[i].name = highest_capacity_malloc(NAME_SIZE);
```

Comment:
As far as I remember, we were talking about example with hashmap where keys are on DRAM and values on PMEM.

In this case, the hashmap structure is on DRAM, but both: keys and values are on PMEM.

Wouldn't it be better to use lowest_latency_malloc for db[i].name? It would reflect: keys on DRAM, values on PMEM.

db[i].data (line above) is ok

Reply:
yes, of course you're right